modules/check_in/app/services/v2/chip/service.rb

```diff
@@ -210,6 +214,29 @@ def appointment_identifiers
           namespace: 'check-in-lorota-v2-cache'
         )
       end
+
+      private
+
+      def demographics_confirmation_payload(confirmed_at:)
+        demographics_up_to_date = check_in_body[:demographics_up_to_date]
+        next_of_kin_up_to_date = check_in_body[:next_of_kin_up_to_date]
+        emergency_contact_up_to_date = check_in_body[:emergency_contact_up_to_date]
+
+        hash = { demographicConfirmations: {} }
+        unless demographics_up_to_date.nil?
+          hash[:demographicConfirmations].merge!({ demographicsNeedsUpdate: !demographics_up_to_date,
+                                                   demographicsConfirmedAt: confirmed_at })
+        end
+        unless next_of_kin_up_to_date.nil?
+          hash[:demographicConfirmations].merge!({ nextOfKinNeedsUpdate: !next_of_kin_up_to_date,
+                                                   nextOfKinConfirmedAt: confirmed_at })
+        end
+        unless emergency_contact_up_to_date.nil?
+          hash[:demographicConfirmations].merge!({ emergencyContactNeedsUpdate: !emergency_contact_up_to_date,
+                                                   emergencyContactConfirmedAt: confirmed_at })
+        end
+        hash
```

Comment:
This is kinda hard for me to follow. I'd suggest:
```
        hsh = {}

        unless check_in_body[:demographics_up_to_date].nil?
          hsh[:demographicsNeedsUpdate] = check_in_body[:demographics_up_to_date] ? false : true
          hsh[:demographicsConfirmedAt] = confirmed_at
        end

        unless check_in_body[:next_of_kin_up_to_date].nil?
          hsh[:nextOfKinNeedsUpdate] = check_in_body[:next_of_kin_up_to_date] ? false : true
          hsh[:nextOfKinConfirmedAt] = confirmed_at
        end

        unless check_in_body[:emergency_contact_up_to_date].nil?
          hsh[:emergencyContactNeedsUpdate] = check_in_body[:emergency_contact_up_to_date] ? false : true
          hsh[:emergencyContactConfirmedAt] = confirmed_at
        end

        {
          demographicConfirmations: hsh
        }
```

Maybe then there's no need to have an extra private method either.

Reply:
As synced offline, added a new method for fixing rubocop failure.  For now disabled rubocop rule as suggested.  Pls find the changes in [commit](https://github.com/department-of-veterans-affairs/vets-api/pull/9336/commits/604e4613bd3bd2d414058dde830ed8cb55b7688f)